runtime/kusama/src/weights/pallet_staking.rs

```diff
@@ -42,181 +44,368 @@ use sp_std::marker::PhantomData;
+	// Storage: Balances Locks (r:1 w:1)
+	// Storage: Staking Ledger (r:0 w:1)
+	// Storage: Staking Payee (r:0 w:1)
+	// Storage: Staking SpanSlash (r:0 w:1)
+	fn reap_stash(s: u32, ) -> Weight {
+		(93_312_000 as Weight)
+			// Standard Error: 1_000
+			.saturating_add((2_226_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(T::DbWeight::get().reads(11 as Weight))
+			.saturating_add(T::DbWeight::get().writes(12 as Weight))
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(s as Weight)))
 	}
-	fn new_era(v: u32, n: u32) -> Weight {
+	// Storage: Staking CounterForNominators (r:1 w:0)
+	// Storage: Staking CounterForValidators (r:1 w:0)
+	// Storage: Staking Validators (r:2 w:0)
+	// Storage: Staking Bonded (r:101 w:0)
+	// Storage: Staking Ledger (r:101 w:0)
+	// Storage: Staking SlashingSpans (r:1 w:0)
+	// Storage: Staking Nominators (r:101 w:0)
+	// Storage: System BlockWeight (r:1 w:1)
+	// Storage: Staking ValidatorCount (r:1 w:0)
+	// Storage: Staking MinimumValidatorCount (r:1 w:0)
+	// Storage: Staking CurrentEra (r:1 w:1)
+	// Storage: Staking HistoryDepth (r:1 w:0)
+	// Storage: Staking ErasStakersClipped (r:0 w:1)
+	// Storage: Staking ErasValidatorPrefs (r:0 w:1)
+	// Storage: Staking ErasStakers (r:0 w:1)
+	// Storage: Staking ErasTotalStake (r:0 w:1)
+	// Storage: Staking ErasStartSessionIndex (r:0 w:1)
+	fn new_era(v: u32, n: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 765_000
-			.saturating_add((309_896_000 as Weight).saturating_mul(v as Weight))
-			// Standard Error: 38_000
-			.saturating_add((53_011_000 as Weight).saturating_mul(n as Weight))
-			.saturating_add(T::DbWeight::get().reads(192 as Weight))
-			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(v as Weight)))
-			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(n as Weight)))
+			// Standard Error: 724_000
+			.saturating_add((300_240_000 as Weight).saturating_mul(v as Weight))
+			// Standard Error: 36_000
+			.saturating_add((49_412_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(10 as Weight))
+			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(v as Weight)))
+			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(n as Weight)))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 			.saturating_add(T::DbWeight::get().writes((3 as Weight).saturating_mul(v as Weight)))
 	}
-	fn get_npos_voters(v: u32, n: u32, s: u32) -> Weight {
+	// Storage: Staking CounterForNominators (r:1 w:0)
+	// Storage: Staking CounterForValidators (r:1 w:0)
+	// Storage: Staking Validators (r:501 w:0)
+	// Storage: Staking Bonded (r:1500 w:0)
+	// Storage: Staking Ledger (r:1500 w:0)
+	// Storage: Staking SlashingSpans (r:21 w:0)
+	// Storage: Staking Nominators (r:1001 w:0)
+	// Storage: System BlockWeight (r:1 w:1)
+	fn get_npos_voters(v: u32, n: u32, s: u32, ) -> Weight {
```

Comment:
hmm how is this benchmark not reading the bags list?

Reply:
just because right now we don't iter over the bags, but instead the nominators map - https://github.com/paritytech/substrate/blob/master/frame/staking/src/pallet/impls.rs#L685 & https://github.com/paritytech/polkadot/blob/e0a53cf5beb87dd21fd50ef1d9be3ac39a0d7488/runtime/common/src/elections.rs#L106-L108